src/vcpkg/dependencies.cpp

```diff
@@ -1307,24 +1312,22 @@ namespace vcpkg::Dependencies
                 }
             };
 
-            std::vector<DepSpec> m_roots;
+            std::vector<DepSpec> m_roots; // the roots of the dependency graph (given in the manifest file)
+            // mapping from portname -> version. "overrides" field in manifest file
             std::map<std::string, Versions::Version> m_overrides;
-            std::map<PackageSpec, PackageNode> m_graph;
+            std::map<PackageSpec, PackageNode> m_graph; // mapping from (portname, triplet) -> what to install
 
             std::pair<const PackageSpec, PackageNode>& emplace_package(const PackageSpec& spec);
 
-            void add_constraint(std::pair<const PackageSpec, PackageNode>& ref,
-                                const Dependency& dep,
-                                const std::string& origin);
-            void add_constraint(std::pair<const PackageSpec, PackageNode>& ref,
-                                const Versions::Version& ver,
-                                const std::string& origin);
-            void add_constraint(std::pair<const PackageSpec, PackageNode>& ref,
-                                const std::string& feature,
-                                const std::string& origin);
-
-            void add_constraint_default_features(std::pair<const PackageSpec, PackageNode>& ref,
-                                                 const std::string& origin);
+            void add_dependency_from(std::pair<const PackageSpec, PackageNode>& ref,
```

Comment:
The reason these are all named "add_constraint" is because they are just that: constraints. These functions do not passively "make this version known to the algorithm"; they have specific semantic effects upon the results.

I'd be fine with seeing them renamed separately, but their names should reflect their semantics. As a suggestion:
- require_dependency
- require_version
- require_feature

or

- add_package_constraint
- add_version_constraint
- add_feature_constraint